cql3/query_options.hh

```diff
@@ -77,6 +78,7 @@ public:
         const std::optional<db::consistency_level> serial_consistency;
         const api::timestamp_type timestamp;
         const service::node_local_only node_local_only;
+        const vector_search::vector_store_client::ann_result ann_result;
```

Comment:
It's not clear to me how ann **results** fits into **query** options. There aren't query options, we don't know them at query time - they are results...

Maybe @avikivity has a better intuition than me here about what this query_options is about.